zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -336,88 +344,79 @@ mod test {
     }
 
     fn mock_tx(
-        value: Option<Word>,
-        gas: Option<u64>,
-        gas_price: Option<Word>,
+        value: Word,
+        gas: Word,
+        gas_price: Word,
         calldata: Vec<u8>,
     ) -> eth_types::Transaction {
         let from = address!("0x00000000000000000000000000000000000000fe");
         let to = address!("0x00000000000000000000000000000000000000ff");
-        let minimal_gas =
-            Word::from(GasCost::TX.as_u64() + 2 * OpcodeId::PUSH32.constant_gas_cost().as_u64());
-        let point_one_ether = Word::from(10u64.pow(17));
-        let two_gwei = Word::from(2_000_000_000);
         eth_types::Transaction {
             from,
             to: Some(to),
-            value: value.unwrap_or(point_one_ether),
-            gas: gas.map(Word::from).unwrap_or(minimal_gas),
-            gas_price: gas_price.or(Some(two_gwei)),
+            value,
+            gas,
+            gas_price: Some(gas_price),
             input: calldata.into(),
             ..Default::default()
         }
     }
 
     #[test]
     fn begin_tx_gadget_simple() {
-        // Transfer 1 ether, successfully
-        test_ok(mock_tx(None, None, None, vec![]), true);
+        // Transfer 0.1 ether, successfully
+        test_ok(
+            mock_tx(*POINT_ONE_ETHER, *MINIMAL_GAS, *TWO_GWEI, vec![]),
+            true,
+        );
 
-        // Transfer 1 ether, tx reverts
-        test_ok(mock_tx(None, None, None, vec![]), false);
+        // Transfer 0.1 ether, tx reverts
+        test_ok(
+            mock_tx(*POINT_ONE_ETHER, *MINIMAL_GAS, *TWO_GWEI, vec![]),
+            false,
+        );
 
         // Transfer nothing with some calldata
         test_ok(
-            mock_tx(None, Some(21086), None, vec![1, 2, 3, 4, 0, 0, 0, 0]),
+            mock_tx(
+                Word::zero(),
+                Word::from(21086),
+                *TWO_GWEI,
+                vec![1, 2, 3, 4, 0, 0, 0, 0],
+            ),
             false,
         );
     }
 
     #[test]
     fn begin_tx_gadget_rand() {
-        let point_one_ether = Word::from(10u8).pow(Word::from(17u8));
+        let random_amount = Word::from_little_endian(&rand_bytes(32)) % *POINT_ONE_ETHER;
+        let random_gas_price = Word::from(rand_range(0..47619047619047u64));
```

Comment:
`47619047619047` comes from <code>10<sup>18</sup> // 21000</code>. Although it could be up to `476190476190476` (<code>10<sup>19</sup> // 21000</code>), but let's use this smaller number to avoid future issue.

Reply:
Was it a typo that the two numbers were different previously (47619047619047 and 476190476190476)?

The sending account always starts with 10^19 wei, so shouldn't the maximum gas price we can pay be 10^19 // 21006 in both cases?

Reply:
> Was it a typo that the two numbers were different previously (47619047619047 and 476190476190476)?

Yes, I think I was trying to update the value (but into an incorrect one) and forget to update both.

> The sending account always starts with 10^19 wei, so shouldn't the maximum gas price we can pay be 10^19 // 21006 in both cases?

Yes, so we should be able to use <code>10<sup>19</sup> // 21006</code> in both cases.

Reply:
ok, i've updated the tests to use 1019 // 21006 in both cases.